src/main/cpp/src/get_json_object.hpp

```diff
@@ -19,10 +19,596 @@
 #include <cudf/strings/string_view.hpp>
 #include <cudf/strings/strings_column_view.hpp>
 
+#include <thrust/optional.h>
+#include <thrust/pair.h>
+#include <thrust/scan.h>
+#include <thrust/tuple.h>
+
 #include <memory>
 
 namespace spark_rapids_jni {
 
+namespace detail {
+
+/**
+ * write JSON style
+ */
+enum class write_style { raw_style, quoted_style, flatten_style };
+
+/**
+ * path instruction type
+ */
+enum class path_instruction_type { subscript, wildcard, key, index, named };
```

Comment:
According to our current style, enum values are `ALL_CAP`.
```suggestion
enum class write_style { RAW_STYLE, ... };
/**
 * path instruction type
 */
enum class path_instruction_type { subscript, wildcard, key, index, named };
```

Reply:
I'll update the enum to upper case after integration test is done.